### DISCUSSION TITLE
Configurable docx frontend

### DISCUSSION THREAD

**Discussion by weilies** [dolanmiu/docx#990] [Ideas / Requests]
I sort of able to write simple DOCX doc. Is it possible to make it "configurable" via angular form?
The intention is to allow admin to create a configurable docx instead of making the whole program inside code?

For a working code below
```
    const doc = new Document({
      sections: [
        {
          headers: {
            default: new Header({
              children: [
                new Paragraph('Header text'),
                new ImageRun({
                  data: blob,
                  transformation: {
                    width: 100,
                    height: 100
                  }
                })
              ]
            })
          },
...
```
Is it possible to make the "new Paragraph...", "new ImageRun..." define via front-end? Coz they are elements under children:[] array. But i don't understand how to loop them? Is it possible?

**Comment by dolanmiu** [comment]
You can if you think about it declaratively

Imagine you have a function, which has inputs, then inside the function, it will create the docx in one single swoop.

```ts
export const createDocument(inputs: MyInputs): Document => {
   const paragraphs = inputs.coolStuff.map((e) => {
       return new Paragraph({
           text: e
       });
   }); 
   
   return new Document({
       sections: [{
           children: [...paragraphs]
       }]
   });
}
```
